lib/ansible/modules/storage/netapp/sf_volume_manager.py

```diff
@@ -0,0 +1,406 @@
+        - locked: No reads or writes are allowed.
+        - replicationTarget: Identify a volume as the target volume for a paired set of volumes. If the volume is not paired, the access status is locked.
+        - If unspecified, the access settings of the clone will be the same as the source.
+
+'''
+
+EXAMPLES = """
+   - name: Create Volume
+     sf_volume_manager:
+       hostname: "{{ solidfire_hostname }}"
+       username: "{{ solidfire_username }}"
+       password: "{{ solidfire_password }}"
+       state: present
+       name: AnsibleVol
+       account_id: 3
+       enable512e: False
+       size: 1
+       size_unit: gb
+
+   - name: Update Volume
+     sf_volume_manager:
+       hostname: "{{ solidfire_hostname }}"
+       username: "{{ solidfire_username }}"
+       password: "{{ solidfire_password }}"
+       state: present
+       name: AnsibleVol
+       account_id: 3
+       new_account_id: 2
+       access: readWrite
+
+   - name: Delete Volume
+     sf_volume_manager:
+       hostname: "{{ solidfire_hostname }}"
+       username: "{{ solidfire_username }}"
+       password: "{{ solidfire_password }}"
+       state: absent
+       name: AnsibleVol
+       account_id: 2
+"""
+
+RETURN = """
+msg:
+    description: Successful creation of Volume
+    returned: success
+    type: string
+    sample: '{"changed": true, "key": value}'
+
+msg:
+    description: Successful update of Volume
+    returned: success
+    type: string
+    sample: '{"changed": true}'
+
+msg:
+    description: Successful removal of Volume
+    returned: success
+    type: string
+    sample: '{"changed": true}'
+
+"""
+
+import logging
+from traceback import format_exc
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.pycompat24 import get_exception
+import ansible.module_utils.netapp as netapp_utils
+
+HAS_SF_SDK = netapp_utils.has_sf_sdk()
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+logging.basicConfig(level=logging.ERROR)
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+
+class SolidFireVolume(object):
+
+    def __init__(self):
+
+        self._size_unit_map = dict(
+
+            # Management GUI displays 1024 ** 3 as 1.1 GB, thus use 1000.
+            bytes=1,
+            b=1,
+            kb=1000,
+            mb=1000 ** 2,
+            gb=1000 ** 3,
+            tb=1000 ** 4,
+            pb=1000 ** 5,
+            eb=1000 ** 6,
+            zb=1000 ** 7,
+            yb=1000 ** 8
+        )
+
+        self.argument_spec = netapp_utils.ontap_sf_host_argument_spec()
+        self.argument_spec.update(dict(
+                state=dict(required=True, choices=['present', 'absent']),
+                name=dict(required=True, type='str'),
+                account_id=dict(required=True, type='int'),
+
+                new_account_id=dict(required=False, type='int', default=None),
+                enable512e=dict(type='bool'),
```

Comment:
Call out default?  If you set a default here, don't forget to update the spec above.